src/main/java/com/amashchenko/maven/plugin/gitflow/AbstractGitFlowMojo.java

```diff
@@ -316,10 +319,20 @@ private String getCurrentProjectOutputTimestamp() throws MojoFailureException {
      * @throws MojoFailureException
      *             If project loading fails.
      */
-    private MavenProject reloadProject(MavenProject project) throws MojoFailureException {
+    private MavenProject reloadProject(final MavenProject project) throws MojoFailureException {
         try {
-            ProjectBuildingResult result = projectBuilder.build(project.getFile(), mavenSession.getProjectBuildingRequest());
-            return result.getProject();
+            List<ProjectBuildingResult> result = projectBuilder.build(
+                    Collections.singletonList(project.getFile()),
+                    true,
+                    mavenSession.getProjectBuildingRequest());
+
+            for (ProjectBuildingResult projectBuildingResult : result) {
+                MavenProject resultProject = projectBuildingResult.getProject();
+                if (!Objects.equals(resultProject.getGroupId(), project.getGroupId())) continue;
```

Comment:
Maybe `resultProject.isExecutionRoot()` can be used instead of comparing ids? Or do you think that root can be somehow different in some cases?

Reply:
Thank you! We didn't know of that property. It works and is what was intended with the ID compares